Fix:

error: call to undeclared function 'lseek'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
error: call to undeclared function 'read'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
error: call to undeclared function 'close'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
error: call to undeclared function 'write'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]

This is fixed in texmaker 5.1.4.
--- texmaker.pro	2022-04-28 00:30:02.000000000 -0500
+++ texmaker.pro	2023-04-30 07:59:36.000000000 -0500
@@ -1148,6 +1148,7 @@
 
 DEFINES += PREFIX=\\\"$${PREFIX}\\\"
 DEFINES +=_FX_CPU_=_FX_X64_
+DEFINES += HAVE_UNISTD_H
 
 
 INCLUDEPATH +=$${QTDIR}/include/
